src/wald_wolfowitz.jl

```diff
@@ -0,0 +1,57 @@
+export 
+    WaldWolfowitzTest
+
+struct WaldWolfowitzTest <: HypothesisTest
+    N_above::Int
+    N_below::Int
+    N_runs::Int
+    μ::Real
```

Comment:
This should be a type parameter to ensure type stability. One parameter is probably enough to all three fields. See how other tests handle it.

Reply:
I don't understand your comments, unfortunately. In other files, I see the use of `abstract type ____ <: HypothesisTest` but I see no need to have an intermediate abstract type rather than a struct.

Reply:
I mean doing `μ::T` and `struct WaldWolfowitzTest{T<:Real} <: HypothesisTest`.

Reply:
Thanks for the clarification, will work on this now